src/combinatorics/set_family/compression/uv.lean

```diff
@@ -230,16 +327,251 @@ end generalized_boolean_algebra
 
 open_locale finset_family
 
-variables [decidable_eq α] {𝒜 : finset (finset α)} {U V A : finset α}
+variables [decidable_eq α] {𝒜 : finset (finset α)} {u v a : finset α}
 
 /-- Compressing a finset doesn't change its size. -/
-lemma card_compress (hUV : U.card = V.card) (A : finset α) : (compress U V A).card = A.card :=
+lemma card_compress (hUV : u.card = v.card) (A : finset α) : (compress u v A).card = A.card :=
 begin
   unfold compress,
   split_ifs,
   { rw [card_sdiff (h.2.trans le_sup_left), sup_eq_union, card_disjoint_union h.1.symm, hUV,
-    add_tsub_cancel_right] },
+      add_tsub_cancel_right] },
   { refl }
 end
 
+/-- UV-compression reduces the size of the shadow of `𝒜` if, for all `x ∈ u` there is `y ∈ v` such
+that `𝒜` is `(u.erase x, v.erase y)`-compressed. This is the key fact about compression for
+Kruskal-Katona. -/
+lemma shadow_compression_subset_compression_shadow (u v : finset α)
+  (huv : ∀ x ∈ u, ∃ y ∈ v, is_compressed (u.erase x) (v.erase y) 𝒜) :
+  ∂ (𝓒 u v 𝒜) ⊆ 𝓒 u v (∂ 𝒜) :=
+begin
+  set 𝒜' := 𝓒 u v 𝒜,
+  suffices H : ∀ s, s ∈ ∂ 𝒜' → s ∉ ∂ 𝒜 →
+    u ⊆ s ∧ disjoint v s ∧ (s ∪ v) \ u ∈ ∂ 𝒜 ∧ (s ∪ v) \ u ∉ ∂ 𝒜',
+  { rintro s hs',
+    rw mem_compression,
+    by_cases hs : s ∈ 𝒜.shadow,
+    { refine or.inl ⟨hs, _⟩,
+      rw compress,
+      split_ifs with huvs,
+      { rcases huvs with ⟨hus, hvs⟩,
+        have hs' := hs',
+        rw [mem_shadow_iff] at hs',
+        rcases hs' with ⟨t, Ht, a, Hat, ta⟩,
+        have hav : a ∉ v := by { 
+          apply not_mem_mono hvs,
+          rw [←ta],
+          exact not_mem_erase a t,
+        },
+        have hvt : v ≤ t := by {
+          rw [←ta] at hvs,
+          exact hvs.trans (erase_subset _ t),
+        },
+        have ht : t ∈ 𝒜 := by {
+          apply mem_of_mem_compression Ht hvt,
+          contrapose!,
+          simp only [bot_eq_empty],
+          intro une,
+          rcases (finset.nonempty_iff_ne_empty.2 une).bex with ⟨x, pf⟩,
+          rcases (huv x pf) with ⟨y, yv, pf⟩,
+          exact ne_empty_of_mem yv,
+        },
+        by_cases hau : a ∈ u,
+        { rcases (huv a hau) with ⟨b, Hbv, Hcomp⟩,
+          rw [is_compressed] at Hcomp,
+          rw [←Hcomp] at ht,
+          have hsb := sup_sdiff_mem_of_mem_compression ht ((finset.erase_subset _ _).trans hvt) sorry, 
+          have : (t ⊔ u.erase a) \ v.erase b = ((s ⊔ u) \ v) ∪ {b} := sorry,
```

Comment:
sorry, maybe i'm too much in "just write code" mode right now, but why is b in `s ⊔ u`?